R/bpvalidate.R

```diff
@@ -124,6 +124,10 @@ setMethod("show", "BPValidate", function(object) {
         i <- vapply(unknown, function(x) {
             !exists(x, envir = env, inherits = FALSE)
         }, logical(1))
+        ## Force evaluation of the known arguments to
+        ## make sure they will be exported correctly
+        known <- unknown[-i]
+        for(i in known) force(env[[i]])
```

Comment:
...but now the for loop changes i? if this fails did your unit test capture this?

Reply:
You are right, I have changed the variable `i` to `obj`

For some reason, I cannot run `devtools::check()` in the Ubuntu system.  I always get the error
```
ERROR in test_rng_lapply_bpredo: Error :   _R_CHECK_LIMIT_CORES_' environment variable detected, BiocParallel
  workers must be <= 2 was (3)
```
This should be related to the recent change in the worker number. However, what puzzled me is that I can manually run the auto-generated check file  `BiocParallel.Rcheck/BiocParallel-Ex.R` with no problem. `R CMD check` is also fine. The only way to reproduce it is to run `devtools::check()` or click the check button in RStudio. Therefore, I have no clue about why I can have this issue. Any thoughts or guess?

Reply:
`devtools::check()` tries to emulate the CRAN build environment, where `_R_CHECK_LIMIT_CORES_` is set. `R CMD check` (and the Bioconductor build system) do not set this variable. It can be set anyway, e.g.,
```
Sys.setenv("_R_CHECK_LIMIT_CORES_" = TRUE)
BiocGenerics::testPackage("BiocParallel", pattern = "test_rng.R")
BiocGenerics::testPackage("BiocParallel")
```
or
```
_R_CHECK_LIMIT_CORES_=TRUE R CMD check .
_R_CHECK_LIMIT_CORES_=TRUE Rscript BiocParallel.Rcheck/BiocParallel-Ex.R
```